core/jvm/src/main/scala/scalaz/zio/App.scala

```diff
@@ -27,12 +25,7 @@ import scala.concurrent.duration.Duration
  */
 trait App extends RTS {
 
-  sealed abstract class ExitStatus extends Serializable with Product
-  object ExitStatus extends Serializable {
-    case class ExitNow(code: Int)                         extends ExitStatus
-    case class ExitWhenDone(code: Int, timeout: Duration) extends ExitStatus
-    case object DoNotExit                                 extends ExitStatus
-  }
+  case class ExitStatus(code: Int) extends Serializable with Product
```

Comment:
Make it final?

Reply:
This doesn't seem to be possible -- classes inside traits (and possibly other classes) can't be declared final.